core/embed/trezorhal/stm32f4/usb/usbd_class_hid.c

```diff
@@ -46,12 +98,6 @@ secbool usb_hid_add(const usb_hid_info_t *info) {
     return secfalse;  // Not enough space in the configuration descriptor
   }
 
-  if ((info->ep_in & USB_EP_DIR_MASK) != USB_EP_DIR_IN) {
```

Comment:
the removal of this check causes misconfiguration if the out endpoint has USB_EP_DIR_IN flag set - previously it would fail here, but now it is propagated. We should perhaps force clear of USB_EP_DIR_MASK bits

Reply:
Now, you can't pass the endpoint number with the `USB_EP_DIR_IN` flag set, as it is not available in the USB driver's public interface. You must and can only specify the endpoint number, ensuring that such a mistake can't happen.

Reply:
You can just pass the endpoint out number as 0x81 for example, you don't need the macro available for that.

Reply:
Then maybe it's better to check for the maximum endpoint number. See the change 680f2de52b857e043b821eb61838b637ffbd2e23.

Reply:
Finally, I found out more appropriate constant, see this change 51e6f541ce7bd526ba5e44239b102b71b4eb25a6